python/samples/kernel-syntax-examples/azure_chat_gpt_api_with_hooks.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from __future__ import annotations
+
+import asyncio
+import os
+
+from semantic_kernel.connectors.ai.open_ai.services.azure_chat_completion import AzureChatCompletion
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.hooks import PostFunctionInvokeContext, PreFunctionInvokeContext
```

Comment:
I did that initially, but it already confused me because FunctionInvoked vs FunctionInvoking is too small of a difference and it made it hard to match the protocol to the context type.